<?xml version="1.0"?>
<launch>
   <arg name="decimation" default="1"/> <!-- 1 -->
   <arg name="debug" default="false"/>
   <arg name="tag" default="false"/>
   <arg name="detect_obj" default="true"/>
   <arg name="detect_box" default="true" />
   <arg name="head_camera" default="false"/>
   <arg name="overhead_camera_file" default="exp_sense_int.yml" />
   <arg name="function_type" default="linear"/>
   <arg name="table_segmentation_threshold" default="0.05"/>

   <arg name="camera_frame" default="camera_rgb_optical_frame"/>
   <arg name="original_image_topic" default="/camera/rgb/image_rect_color"/>
   <arg name="image_topic" default="$(arg original_image_topic)_drop"/>
   <arg name="cloud_topic" default="/camera/depth_registered/points"/>
   <arg name="ref_frame" default="/base" />

   <arg name="exp_rec" default="ltm"/> <!--ltm/mot-->
   <arg name="exp_type" default="regular"/>
   <arg name="config_time" default="20" />
   <arg name="fixed_height" default="-0.04" />
   <arg name="super_throw" default="false" />

   <arg name="cal_xmin" default="0.32"/>
   <arg name="cal_xmax" default="1.34"/>
   <arg name="cal_ymin" default="-1.10"/>
   <arg name="cal_ymax" default="1.10"/>
   <arg name="cal_ang" default="0.98125"/>

   <arg name="grab_param_file" default="grab_parameters.yml"/>
   <arg name="throw_param_file" default="throw_parameters.yml"/>
   <arg name="motiven_param_file" default="motiven_parameters.yml"/>
   <arg name="robobo_param_file" default="robobo_parameters.yml"/>

   <node name="drop_node" pkg="topic_tools" type="drop" args="$(arg original_image_topic) 3 4 $(arg image_topic)"/>

   <include file="$(find mdb_robots_detection)/launch/head_aruco_marker.launch" if="$(arg tag)"/> 

   <node name="exp_track" pkg="mdb_robots_detection" type="exp_track.py" required="true" output="screen" unless="$(arg debug)">
       <remap from="/image_color" to="$(arg image_topic)" />
       <param name="exp_rec" type="str" value="$(arg exp_rec)" />
       <param name="head_camera" type="bool" value="$(arg head_camera)"/> 
       <param name="table_segmentation_threshold" type="double" value="$(arg table_segmentation_threshold)"/>

	   <rosparam param="c_rgb">[75, 157, 143]</rosparam>
	   <rosparam param="b_rgb">[66, 55, 127]</rosparam>
   </node>

   <node name="ball_pos" pkg="mdb_robots_detection" type="obj_pos_int.py" required="true" output="screen" if="$(arg detect_obj)">
       <param name="obj_type" type="str" value="ball"/>
	   <param name="camera_frame" type="str" value="$(arg camera_frame)"/>
	   <param name="overhead_camera_file" type="str" value="$(arg overhead_camera_file)" />
       <param name="function_type" type="str" value="$(arg function_type)"/>

   </node>

	<node name="box_pos" pkg="mdb_robots_detection" type="obj_pos_int.py" required="true" output="screen" if="$(arg detect_box)">
       <param name="obj_type" type="str" value="box"/>
	   <param name="camera_frame" type="str" value="$(arg camera_frame)"/>
	   <param name="overhead_camera_file" type="str" value="$(arg overhead_camera_file)" />
       <param name="function_type" type="str" value="$(arg function_type)"/>
   </node>

	<node name="robobo_pos" pkg="mdb_robots_detection" type="obj_pos_int.py" required="true" output="screen" if="$(arg tag)">
       <param name="obj_type" type="str" value="robobo"/>
	   <param name="camera_frame" type="str" value="$(arg camera_frame)"/>
	   <param name="overhead_camera_file" type="str" value="$(arg overhead_camera_file)" />
       <param name="function_type" type="str" value="$(arg function_type)"/>
	</node>
   <node name="get_joints_state_server" pkg="mdb_robots_policies"  type="get_joints_state.py" required="true" output="log"/>
   <node name="get_end_state_server" pkg="mdb_robots_policies"  type="get_end_state.py" required="true" output="log"/>
   <node name="get_gripper_state_server" pkg="mdb_robots_policies" type="get_gripper_state.py" required="true" output="log"/>
   <node name="get_head_state" pkg="mdb_robots_policies" type="get_head_state.py" required="true" output="log"/>

   <node name="exp_scene_server" pkg="mdb_robots_policies" type="exp_scene.py" required="true" output="screen"/>

   <node name="open_grip_left" pkg="mdb_robots_policies" type="open_grip.py" required="true" output="screen">
     <param name="arm" type="str" value="left" />
   </node>

   <node name="open_grip_right" pkg="mdb_robots_policies" type="open_grip.py" required="true" output="screen">
     <param name="arm" type="str" value="right" />
   </node>

   <node name="baxter_policies" pkg="mdb_robots_policies" type="baxter_policies.py" required="true" output="screen">
     <param name="exp_rec" type="str" value="$(arg exp_rec)" />	
	 <param name="super_throw" type="bool" value="$(arg super_throw)"/>

	 <param name="cal_xmin" type="double" value="$(arg cal_xmin)"/>
	 <param name="cal_xmax" type="double" value="$(arg cal_xmax)"/>
	 <param name="cal_ymin" type="double" value="$(arg cal_ymin)"/>
	 <param name="cal_ymax" type="double" value="$(arg cal_ymax)"/>
     <param name="cal_ang" type="double" value="$(arg cal_ang)"/>

	 <param name="grab_param_file" type="str" value="$(arg grab_param_file)"/>
	 <param name="throw_param_file" type="str" value="$(arg throw_param_file)"/>
	 <param name="motiven_param_file" type="str" value="$(arg motiven_param_file)"/>
	 <param name="robobo_param_file" type="str" value="$(arg robobo_param_file)"/>

   </node>

   <node name="exp_ltm" pkg="mdb_robots_experiments" type="exp_core.py" required="true" output="screen">
	 <param name="exp_type" type="str" value="$(arg exp_type)" />
	 <param name="config_time" type="int" value="$(arg config_time)" />
	 <param name="fixed_height" type="double" value="$(arg fixed_height)" />
	 <param name="super_throw" type="bool" value="$(arg super_throw)"/>
     <param name="exp_rec" type="str" value="$(arg exp_rec)" />	
   </node>

   <node name="exp_helper" pkg="mdb_robots_experiments" type="exp_helper.py" required="true" output="screen"/> 
</launch>
